In [1]:
import pandas as pd
import numpy as np

In [2]:
cd data

/Users/alexiscarlier/Desktop/S2/Applied_Econometrics/data


In [3]:
df_before_78 = pd.read_csv('before_78_1.csv')

/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


## Deleting data that was incorrectly entered by the surveyors

In [4]:
clean_before_78_1 = df_before_78[pd.notnull(df_before_78['parcel'])]

In [5]:
before_78_2 = pd.read_csv('before_78_2.csv')

/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [6]:
before_78_2.head()

Unnamed: 0 home_id         parcel   amount  month    year usage_type unit  \
0           1  167394  13451-002-000  10150.0   12.0  2000.0       CO2e    g   
1           2  167394  13451-002-000   9425.0    1.0  2001.0       CO2e    g   
2           3  167394  13451-002-000  10150.0    2.0  2001.0       CO2e    g   
3           4  167394  13451-002-000   7250.0    3.0  2001.0       CO2e    g   
4           5  167394  13451-002-000   9425.0    4.0  2001.0       CO2e    g   

   gco2e_coefficient  subdivision_id               subdivision  apartment_id  \
0                1.0          2643.0  WILSONS S/D MILLER BLOCK           NaN   
1                1.0          2643.0  WILSONS S/D MILLER BLOCK           NaN   
2                1.0          2643.0  WILSONS S/D MILLER BLOCK           NaN   
3                1.0          2643.0  WILSONS S/D MILLER BLOCK           NaN   
4                1.0          2643.0  WILSONS S/D MILLER BLOCK           NaN   

   apartment  squarefeet    zip  builtin  beds  baths  stories        family  
0        NaN        2486  32602     1975     4    2.0        2  multi family  
1        NaN        2486  32602     1975     4    2.0        2  multi family  
2        NaN        2486  32602     1975     4    2.0        2  multi family  
3        NaN        2486  32602     1975     4    2.0        2  multi family  
4        NaN        2486  32602     1975     4    2.0        2  multi family

In [7]:
clean_before_78_2 = before_78_2[pd.notnull(before_78_2['parcel'])]

In [8]:
len(set(list(clean_before_78_2['home_id']))) 

61

In [9]:
len(set(list(clean_before_78_1['home_id'])))

119

In [10]:
clean_before_78 = pd.concat([clean_before_78_1, clean_before_78_2], ignore_index=True)

In [11]:
len(set(list(clean_before_78['home_id'])))

180

In [12]:
df_1974 = pd.read_csv('1974.csv')

In [13]:
len(set(list(df_1974['home_id']))) + 180

228

## This is the correct(ish) version of the before 1978 data

In [14]:
clean_before_78 = pd.concat([clean_before_78, df_1974], ignore_index=True)

In [15]:
len(set(list(clean_before_78['home_id'])))

228

## Now for the post 1978 data

In [16]:
post_78 = pd.read_csv('datafrom79.csv')

/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


### Deleting (more!) data that was incorrectly entered by the surveyors

In [17]:
clean_79 = post_78[pd.notnull(post_78['parcel'])]

In [18]:
df_1982 = pd.read_csv('1982.csv')

/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [19]:
df_1982 = df_1982[pd.notnull(df_1982['parcel'])]

In [20]:
len(set(list(df_1982['home_id'])))

49

In [21]:
clean_post_78 = pd.concat([clean_79, df_1982], ignore_index=True)

/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=True'.

To retain the current behavior and silence the warning, pass sort=False

  """Entry point for launching an IPython kernel.


#### One of the houses is incorrectly entered in the database as being built in both 1977 and 1982...

In [22]:
def intersection(lst1, lst2): 
    return list(set(lst1) & set(lst2)) 

In [23]:
intersection(list(clean_post_78['home_id']), clean_before_78['home_id'])

['162311']

In [24]:
clean_post_78[clean_post_78['home_id'] == '162311']

Unnamed: 0     amount apartment  apartment_id  baths  beds  builtin  \
114595     14329.0   692850.0       NaN           NaN    2.0     3     1982   
114596     14330.0   729700.0       NaN           NaN    2.0     3     1982   
114597     14331.0   991025.0       NaN           NaN    2.0     3     1982   
114598     14332.0  1367400.0       NaN           NaN    2.0     3     1982   
114599     14333.0  1161600.0       NaN           NaN    2.0     3     1982   
114600     14334.0   853525.0       NaN           NaN    2.0     3     1982   
114601     14335.0   756800.0       NaN           NaN    2.0     3     1982   
114602     14336.0   812025.0       NaN           NaN    2.0     3     1982   
114603     14337.0  1411800.0       NaN           NaN    2.0     3     1982   
114604     14338.0  1520700.0       NaN           NaN    2.0     3     1982   
114605     14339.0  1235450.0       NaN           NaN    2.0     3     1982   
114606     14340.0  1132450.0       NaN           NaN    2.0     3     1982   
114607     14341.0   571625.0       NaN           NaN    2.0     3     1982   
114608     14342.0   700500.0       NaN           NaN    2.0     3     1982   
114609     14343.0  1061675.0       NaN           NaN    2.0     3     1982   
114610     14344.0  1609675.0       NaN           NaN    2.0     3     1982   
114611     14345.0   742975.0       NaN           NaN    2.0     3     1982   
114612     14346.0   511925.0       NaN           NaN    2.0     3     1982   
114613     14347.0   499225.0       NaN           NaN    2.0     3     1982   
114614     14348.0   749575.0       NaN           NaN    2.0     3     1982   
114615     14349.0  1091600.0       NaN           NaN    2.0     3     1982   
114616     14350.0  1584150.0       NaN           NaN    2.0     3     1982   
114617     14351.0  1212325.0       NaN           NaN    2.0     3     1982   
114618     14352.0   750225.0       NaN           NaN    2.0     3     1982   
114619     14353.0   530100.0       NaN           NaN    2.0     3     1982   
114620     14354.0   507875.0       NaN           NaN    2.0     3     1982   
114621     14355.0   471875.0       NaN           NaN    2.0     3     1982   
114622     14356.0  1010875.0       NaN           NaN    2.0     3     1982   
114623     14357.0   664375.0       NaN           NaN    2.0     3     1982   
114624     14358.0   462075.0       NaN           NaN    2.0     3     1982   
...            ...        ...       ...           ...    ...   ...      ...   
115640     15374.0        4.0       NaN           NaN    2.0     3     1982   
115641     15375.0        4.0       NaN           NaN    2.0     3     1982   
115642     15376.0        4.0       NaN           NaN    2.0     3     1982   
115643     15377.0        4.0       NaN           NaN    2.0     3     1982   
115644     15378.0        3.0       NaN           NaN    2.0     3     1982   
115645     15379.0        3.0       NaN           NaN    2.0     3     1982   
115646     15380.0        3.0       NaN           NaN    2.0     3     1982   
115647     15381.0        4.0       NaN           NaN    2.0     3     1982   
115648     15382.0        2.0       NaN           NaN    2.0     3     1982   
115649     15383.0        3.0       NaN           NaN    2.0     3     1982   
115650     15384.0        3.0       NaN           NaN    2.0     3     1982   
115651     15385.0        3.0       NaN           NaN    2.0     3     1982   
115652     15386.0        3.0       NaN           NaN    2.0     3     1982   
115653     15387.0        4.0       NaN           NaN    2.0     3     1982   
115654     15388.0        4.0       NaN           NaN    2.0     3     1982   
115655     15389.0        4.0       NaN           NaN    2.0     3     1982   
115656     15390.0        3.0       NaN           NaN    2.0     3     1982   
115657     15391.0        4.0       NaN           NaN    2.0     3     1982   
115658     15392.0        4.0       NaN           NaN    2

In [25]:
clean_before_78[clean_before_78['home_id'] == '162311']

Unnamed: 0 home_id         parcel     amount  month    year usage_type  \
12397       13443  162311  06528-029-000   692850.0   10.0  1999.0       CO2e   
12398       13444  162311  06528-029-000   729700.0   11.0  1999.0       CO2e   
12399       13445  162311  06528-029-000   991025.0   12.0  1999.0       CO2e   
12400       13446  162311  06528-029-000  1367400.0    1.0  2000.0       CO2e   
12401       13447  162311  06528-029-000  1161600.0    2.0  2000.0       CO2e   
12402       13448  162311  06528-029-000   853525.0    3.0  2000.0       CO2e   
12403       13449  162311  06528-029-000   756800.0    4.0  2000.0       CO2e   
12404       13450  162311  06528-029-000   812025.0    5.0  2000.0       CO2e   
12405       13451  162311  06528-029-000  1411800.0    6.0  2000.0       CO2e   
12406       13452  162311  06528-029-000  1520700.0    7.0  2000.0       CO2e   
12407       13453  162311  06528-029-000  1235450.0    8.0  2000.0       CO2e   
12408       13454  162311  06528-029-000  1132450.0    9.0  2000.0       CO2e   
12409       13455  162311  06528-029-000   571625.0   10.0  2000.0       CO2e   
12410       13456  162311  06528-029-000   700500.0   11.0  2000.0       CO2e   
12411       13457  162311  06528-029-000  1061675.0   12.0  2000.0       CO2e   
12412       13458  162311  06528-029-000  1609675.0    1.0  2001.0       CO2e   
12413       13459  162311  06528-029-000   742975.0    2.0  2001.0       CO2e   
12414       13460  162311  06528-029-000   511925.0    3.0  2001.0       CO2e   
12415       13461  162311  06528-029-000   499225.0    4.0  2001.0       CO2e   
12416       13462  162311  06528-029-000   749575.0    5.0  2001.0       CO2e   
12417       13463  162311  06528-029-000  1091600.0    6.0  2001.0       CO2e   
12418       13464  162311  06528-029-000  1584150.0    7.0  2001.0       CO2e   
12419       13465  162311  06528-029-000  1212325.0    8.0  2001.0       CO2e   
12420       13466  162311  06528-029-000   750225.0    9.0  2001.0       CO2e   
12421       13467  162311  06528-029-000   530100.0   10.0  2001.0       CO2e   
12422       13468  162311  06528-029-000   507875.0   11.0  2001.0       CO2e   
12423       13469  162311  06528-029-000   471875.0   12.0  2001.0       CO2e   
12424       13470  162311  06528-029-000  1010875.0    1.0  2002.0       CO2e   
12425       13471  162311  06528-029-000   664375.0    2.0  2002.0       CO2e   
12426       13472  162311  06528-029-000   462075.0    3.0  2002.0       CO2e   
...           ...     ...            ...        ...    ...     ...        ...   
13442       14488  162311  06528-029-000        4.0    7.0  2016.0      water   
13443       14489  162311  06528-029-000        4.0    8.0  2016.0      water   
13444       14490  162311  06528-029-000        4.0    9.0  2016.0      water   
13445       14491  162311  06528-029-000        4.0   10.0  2016.0      water   
13446       14492  162311  06528-029-000        3.0   11.0  2016.0      water   
13447       14493  162311  06528-029-000        3.0   12.0  2016.0      water   
13448       14494  162311  06528-029-000        3.0    1.0  2017.0      water   
13449       14495  162311  06528-029-000        4.0    2.0  2017.0      water   
13450       14496  162311  06528-029-000        2.0    3.0  2017.0      water   
13451       14497  162311  06528-029-000        3.0    4.0  2017.0      water   
13452       14498  162311  06528-029-000        3.0    5.0  2017.0      water   
13453       14499  162311  06528-029-000        3.0    6.0  2017.0      water   
13454       14500  162311  06528-029-000        3.0    7.0  2017.0      water   
13455       14501  162311  06528-029-000        4.0    8.0  2017.0      water   
13456       14502  162311  06528-029-000        4.0    9.0  2017.0      water   
13457       14503  162311  06528-029-000        4.0   10.0  2017.0      water   
13458       14504  162311  06528-029-000        3.0   11.0  2017.0      water   
13459       14505  162311  06528-029-0

#### So I'll delete it from each of the Dfs

In [26]:
clean_before_78 = clean_before_78[clean_before_78['home_id'] != '162311']

# df = df[df.line_race != 0]

In [27]:
len(set(list(clean_before_78['home_id'])))

227

In [28]:
clean_post_78 = clean_post_78[clean_post_78['home_id'] != '162311']

In [29]:
len(set(list(clean_post_78['home_id'])))

183

Merging the two dataframes. We have 184 houses in the post-1978 and 228 in the pre-1978 data bases

In [30]:
all_houses = pd.concat([clean_before_78, clean_post_78], sort=True)

In [31]:
all_houses.head()

Unnamed: 0    amount apartment  apartment_id  baths  beds  builtin  \
0         1.0    1450.0       NaN           NaN    2.0     3     1977   
1         2.0    1450.0       NaN           NaN    2.0     3     1977   
2         3.0  325200.0       NaN           NaN    2.0     3     1977   
3         4.0  288200.0       NaN           NaN    2.0     3     1977   
4         5.0  278225.0       NaN           NaN    2.0     3     1977   

          family  gco2e_coefficient home_id  month         parcel  squarefeet  \
0  single family                1.0  158340   10.0  11515-000-000         804   
1  single family                1.0  158340   11.0  11515-000-000         804   
2  single family                1.0  158340   12.0  11515-000-000         804   
3  single family                1.0  158340    1.0  11515-000-000         804   
4  single family                1.0  158340    2.0  11515-000-000         804   

   stories subdivision  subdivision_id unit usage_type    year    zip  
0        1         NaN             NaN    g       CO2e  1999.0  32601  
1        1         NaN             NaN    g       CO2e  1999.0  32601  
2        1         NaN             NaN    g       CO2e  1999.0  32601  
3        1         NaN             NaN    g       CO2e  2000.0  32601  
4        1         NaN             NaN    g       CO2e  2000.0  32601

#### The new dataframe has the correct number of houses as it's the sum of the number of houses in each list

In [32]:
len(set(list(all_houses['home_id'])))

410

#### Removing the excess indexing column

In [33]:
all_houses = all_houses.drop(['Unnamed: 0'], axis=1)

In [34]:
all_houses.to_pickle("./all_houses.pkl")

## Loading the electricity and natural gas prices to add to dfs

In [35]:
electricty = pd.read_excel('avgprice_annual_electricity.xlsx')

In [36]:
natural_gas = pd.read_excel('natural_gas_price.xls')

Reducing the electricty data to just include that from Florida

In [37]:
electricity = electricty[electricty['State'] == 'FL']

## Note that the unit for the electricity measurements is cents/killowatthour

In [38]:
electricity.head()

Year State Industry Sector Category  Residential  Commercial  Industrial  \
9    2017    FL  Total Electric Industry        11.61        9.35        7.83   
61   2017    FL   Full-Service Providers        11.61        9.35        7.83   
176  2016    FL  Total Electric Industry        10.98        8.90        7.69   
228  2016    FL   Full-Service Providers        10.98        8.90        7.69   
343  2015    FL  Total Electric Industry        11.58        9.50        8.22   

     Transportation  Other  Total  
9              8.62    NaN  10.42  
61             8.62    NaN  10.42  
176            8.32    NaN   9.91  
228            8.32    NaN   9.91  
343            8.92    NaN  10.49

#### Deleting all the electricity stuff I don't need

In [39]:
electricity = electricity[electricity['Industry Sector Category'] == 'Total Electric Industry']
electricity.head()

cleaned_electricity = electricity.drop(['State', 'Industry Sector Category', 'Industrial', 'Commercial', 'Transportation', 'Other', 'Total'], axis=1)

### Idea: merge the dataframes on year. First, change the year column so it's the same in both, then merge

In [40]:
cleaned_electricity = cleaned_electricity.rename(columns={'Year': 'year'})

In [41]:
all_houses = pd.merge(all_houses, cleaned_electricity, on='year', how='outer')

In [42]:
all_houses = all_houses.rename(columns={'Residential': 'avg electricity price cents/kwh'})

In [43]:
all_houses

amount apartment  apartment_id  baths  beds  builtin  \
0          1450.0       NaN           NaN    2.0     3     1977   
1          1450.0       NaN           NaN    2.0     3     1977   
2        325200.0       NaN           NaN    2.0     3     1977   
3           350.0       NaN           NaN    2.0     3     1977   
4           350.0       NaN           NaN    2.0     3     1977   
5             2.0       NaN           NaN    2.0     3     1977   
6             2.0       NaN           NaN    2.0     3     1977   
7             2.0       NaN           NaN    2.0     3     1977   
8        848600.0       NaN           NaN    2.0     3     1977   
9       1091650.0       NaN           NaN    2.0     3     1977   
10          858.0       NaN           NaN    2.0     3     1977   
11          890.0       NaN           NaN    2.0     3     1977   
12         1121.7       NaN           NaN    2.0     3     1977   
13         2325.7       NaN           NaN    2.0     3     1977   
14            9.0       NaN           NaN    2.0     3     1977   
15           49.0       NaN           NaN    2.0     3     1977   
16           10.0       NaN           NaN    2.0     3     1977   
17           12.0       NaN           NaN    2.0     3     1977   
18        62900.0       NaN           NaN    1.0     2     1978   
19        75850.0       NaN           NaN    1.0     2     1978   
20       106375.0       NaN           NaN    1.0     2     1978   
21           68.0       NaN           NaN    1.0     2     1978   
22           82.0       NaN           NaN    1.0     2     1978   
23          115.0       NaN           NaN    1.0     2     1978   
24           68.0       NaN           NaN    1.0     2     1978   
25           82.0       NaN           NaN    1.0     2     1978   
26          115.0       NaN           NaN    1.0     2     1978   
27       740375.0       NaN           NaN    2.5     4     1977   
28       717825.0       NaN           NaN    2.5     4     1977   
29          722.0       NaN           NaN    2.5     4     1977   
...           ...       ...           ...    ...   ...      ...   
355505    95400.0       NaN           NaN    1.0     3     1979   
355506   106000.0       NaN           NaN    1.0     3     1979   
355507   100700.0       NaN           NaN    1.0     3     1979   
355508   100700.0       NaN           NaN    1.0     3     1979   
355509   185500.0       NaN           NaN    1.0     3     1979   
355510   259700.0       NaN           NaN    1.0     3     1979   
355511     2490.5       NaN           NaN    1.0     3     1979   
355512     2256.1       NaN           NaN    1.0     3     1979   
355513     2637.0       NaN           NaN    1.0     3     1979   
355514     2197.5       NaN           NaN    1.0     3     1979   
355515     1113.4       NaN           NaN    1.0     3     1979   
355516      527.4       NaN           NaN    1.0     3     1979   
355517      527.4       NaN           NaN    1.0     3     1979   
355518      586.0       NaN           NaN    1.0     3     1979   
355519      556.7       NaN           NaN    1.0     3     1979   
355520      556.7       NaN           NaN    1.0     3     1979   
355521     1025.5       NaN           NaN    1.0     3     1979   
355522     1435.7       NaN           NaN    1.0     3     1979   
355523       85.0       NaN           NaN    1.0     3     1979   
355524       77.0       NaN           NaN    1.0     3     1979   
355525       90.0       NaN           NaN    1.0     3     1979   
355526       75.0       NaN           NaN    1.0     3     1979   
355527       38.0       NaN           NaN    1.0     3     1979   
355528       18.0       NaN           NaN    1.0     3     1979   
355529       18.0       NaN           NaN    1.0     3     1979   
355530       20.0       NaN           NaN    1.0     3     1979   
355531       19.0       NaN           NaN    1.0     3     1979   
355532       19.0       NaN           NaN    1.0     3    

#### Same process now for natural gas

In [44]:
natural_gas.head()

Date  \
0 1989-01-15   
1 1989-02-15   
2 1989-03-15   
3 1989-04-15   
4 1989-05-15   

   Florida Price of Natural Gas Delivered to Residential Consumers (Dollars per Thousand Cubic Feet)  
0                                               7.26                                                  
1                                               7.55                                                  
2                                               7.12                                                  
3                                               7.92                                                  
4                                               8.91

In [45]:
natural_gas = natural_gas.rename(columns={'Florida Price of Natural Gas Delivered to Residential Consumers (Dollars per Thousand Cubic Feet)': 'avg natural gas price $/thousand cubic feet'})

In [46]:
natural_gas['Date'] = natural_gas['Date'].apply(lambda x: x.year)

In [47]:
natural_gas = natural_gas.rename(columns={'Date': 'year'})

In [48]:
all_houses = pd.merge(all_houses, natural_gas, on='year', how='outer')

In [49]:
all_houses.head()

amount apartment  apartment_id  baths  beds  builtin         family  \
0  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
1  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
2  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
3  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
4  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   

   gco2e_coefficient home_id  month  \
0                1.0  158340   10.0   
1                1.0  158340   10.0   
2                1.0  158340   10.0   
3                1.0  158340   10.0   
4                1.0  158340   10.0   

                      ...                      squarefeet  stories  \
0                     ...                           804.0      1.0   
1                     ...                           804.0      1.0   
2                     ...                           804.0      1.0   
3                     ...                           804.0      1.0   
4                     ...                           804.0      1.0   

   subdivision subdivision_id  unit usage_type    year      zip  \
0          NaN            NaN     g       CO2e  1999.0  32601.0   
1          NaN            NaN     g       CO2e  1999.0  32601.0   
2          NaN            NaN     g       CO2e  1999.0  32601.0   
3          NaN            NaN     g       CO2e  1999.0  32601.0   
4          NaN            NaN     g       CO2e  1999.0  32601.0   

   avg electricity price cents/kwh  \
0                             7.73   
1                             7.73   
2                             7.73   
3                             7.73   
4                             7.73   

   avg natural gas price $/thousand cubic feet  
0                                         9.75  
1                                        10.71  
2                                        10.15  
3                                        11.14  
4                                        12.46  

[5 rows x 21 columns]

### Adding a proxy for revenue

In [50]:
all_houses.head()

amount apartment  apartment_id  baths  beds  builtin         family  \
0  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
1  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
2  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
3  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   
4  1450.0       NaN           NaN    2.0   3.0   1977.0  single family   

   gco2e_coefficient home_id  month  \
0                1.0  158340   10.0   
1                1.0  158340   10.0   
2                1.0  158340   10.0   
3                1.0  158340   10.0   
4                1.0  158340   10.0   

                      ...                      squarefeet  stories  \
0                     ...                           804.0      1.0   
1                     ...                           804.0      1.0   
2                     ...                           804.0      1.0   
3                     ...                           804.0      1.0   
4                     ...                           804.0      1.0   

   subdivision subdivision_id  unit usage_type    year      zip  \
0          NaN            NaN     g       CO2e  1999.0  32601.0   
1          NaN            NaN     g       CO2e  1999.0  32601.0   
2          NaN            NaN     g       CO2e  1999.0  32601.0   
3          NaN            NaN     g       CO2e  1999.0  32601.0   
4          NaN            NaN     g       CO2e  1999.0  32601.0   

   avg electricity price cents/kwh  \
0                             7.73   
1                             7.73   
2                             7.73   
3                             7.73   
4                             7.73   

   avg natural gas price $/thousand cubic feet  
0                                         9.75  
1                                        10.71  
2                                        10.15  
3                                        11.14  
4                                        12.46  

[5 rows x 21 columns]

### Deleting unecessary columns

In [51]:
all_houses = all_houses.drop(['parcel', 'subdivision', 'subdivision_id', 'apartment_id', 'unit'], 1)

### Creating a correlation matrix

In [52]:
all_houses.corr()

amount     baths      beds  \
amount                                       1.000000  0.084702  0.077065   
baths                                        0.084702  1.000000  0.708024   
beds                                         0.077065  0.708024  1.000000   
builtin                                      0.000847 -0.130070 -0.164474   
gco2e_coefficient                           -0.376370  0.036399  0.078015   
month                                        0.002825 -0.001930 -0.003660   
squarefeet                                   0.096691  0.631693  0.607237   
stories                                      0.045105  0.291800  0.124774   
year                                        -0.050144 -0.055526 -0.095537   
zip                                          0.005424  0.123296  0.060267   
avg electricity price cents/kwh             -0.040786 -0.064736 -0.101997   
avg natural gas price $/thousand cubic feet -0.019327 -0.029302 -0.049119   

                                              builtin  gco2e_coefficient  \
amount                                       0.000847          -0.376370   
baths                                       -0.130070           0.036399   
beds                                        -0.164474           0.078015   
builtin                                      1.000000          -0.076944   
gco2e_coefficient                           -0.076944           1.000000   
month                                        0.001467           0.006047   
squarefeet                                  -0.049919           0.032855   
stories                                      0.063037          -0.050031   
year                                         0.024066          -0.045557   
zip                                         -0.026524           0.008943   
avg electricity price cents/kwh              0.029174          -0.025071   
avg natural gas price $/thousand cubic feet  0.012668          -0.025284   

                                                month  squarefeet   stories  \
amount                                       0.002825    0.096691  0.045105   
baths                                       -0.001930    0.631693  0.291800   
beds                                        -0.003660    0.607237  0.124774   
builtin                                      0.001467   -0.049919  0.063037   
gco2e_coefficient                            0.006047    0.032855 -0.050031   
month                                        1.000000   -0.000597  0.001543   
squarefeet                                  -0.000597    1.000000  0.270225   
stories                                      0.001543    0.270225  1.000000   
year                                        -0.008661   -0.040806  0.017869   
zip                                          0.000589    0.058182 -0.023154   
avg electricity price cents/kwh             -0.017285   -0.048181  0.020391   
avg natural gas price $/thousand cubic feet -0.016963   -0.023169  0.009094   

                                                 year       zip  \
amount                                      -0.050144  0.005424   
baths                                       -0.055526  0.123296   
beds                                        -0.095537  0.060267   
builtin                                      0.024066 -0.026524   
gco2e_coefficient                           -0.045557  0.008943   
month                                       -0.008661  0.000589   
squarefeet                                  -0.040806  0.058182   
stories                                      0.017869 -0.023154   
year                                         1.000000  0.001567   
zip                                          0.001567  1.000000   
avg electricity price cents/kwh              0.813278  0.003850   
avg natural gas price $/thousand cubic feet  0.502022  0.004217   

                                             avg electricity price cents/kwh  \
amount                                                             -0.040786